src/Infrastructure/Persistence/Repository/ApplicationDbRepository.cs

```diff
@@ -2,17 +2,24 @@
 using Ardalis.Specification.EntityFrameworkCore;
 using FSH.WebApi.Application.Common.Persistence;
 using FSH.WebApi.Domain.Common.Contracts;
+using FSH.WebApi.Infrastructure.Common;
 using FSH.WebApi.Infrastructure.Persistence.Context;
+using FSH.WebApi.Infrastructure.Persistence.Specification.Evaluators;
 using Mapster;
+using Microsoft.Extensions.Options;
 
 namespace FSH.WebApi.Infrastructure.Persistence.Repository;
 
 // Inherited from Ardalis.Specification's RepositoryBase<T>
 public class ApplicationDbRepository<T> : RepositoryBase<T>, IReadRepository<T>, IRepository<T>
     where T : class, IAggregateRoot
 {
-    public ApplicationDbRepository(ApplicationDbContext dbContext)
-        : base(dbContext)
+    public ApplicationDbRepository(ApplicationDbContext dbContext, IOptions<DatabaseSettings> dbSettings)
+        : base(
+            dbContext,
+            dbSettings.Value.DBProvider == DbProviderKeys.Npgsql
+                ? FSHSpecificationEvaluator.NpgsqlInstance
+                : FSHSpecificationEvaluator.DefaultInstance)
```

Comment:
Hmmm starting to wonder again... 

Maybe we should delegate to `SpecificationEvaluator.Default` instead after all? As long as there isn't anything different?

The naming will have to change again then though... 
NpgSqlSpecificationEvaluator and NpgSqlSearchEvaluator

I think we (I ;-)) over-complicated it a bit now... sorry for that.

Now I think it's better to "keep it simple". No parameter in the SearchEvaluator... If we ever need another evaluator that does something similar we can always re-factor a base class out of it then... but until then "YAGNI" ;-)

Reply:
On the one hand, it would be nice to have it under the Repository\NpgSql directory as this code is only used by NpgSql provider and people who don't use NpgSql would know they can safely remove it, but on the other hand, this code is related to the specification pattern implementation that's why I followed the same namespace/folder structure like in Ardalis.Specification package.

In my opinion, the whole thing should be handled by the Specification package but unfortunately, they don't want to add it...
I'm worried about what will happen if we update the Specification package in the feature and there will be some changes in their default specification evaluator then we will have to remember to update our custom evaluator
So we have to decide if we want to keep all this extra code in the boilerplate or if we want to put this in docs only as an example.

Reply:
Yeah, but the repository itself is also part of ardalis specification, so putting it under there would make sense, no? Maybe add a readme or code comments which link to the actual code in the specifications repository where this is copied from, so we can more easily revisit in the future, when it will become necessary?

Once this is merged, we can show our implementation to the guys from ardalis specification and they can maybe open up more of the internal types, and maybe even make the "like function" parameterizable, so we can delete some code again ;-)